### DISCUSSION TITLE
https://www.spsanderson.com/steveondata/posts/2024-10-22/

### DISCUSSION THREAD

**Discussion by giscus** [spsanderson/steveondata#12] [Comments]
# https://www.spsanderson.com/steveondata/posts/2024-10-22/

Master list manipulation in R using base loops and purrr. Learn efficient techniques with practical examples for beginners. Boost your data analysis skills today!

https://www.spsanderson.com/steveondata/posts/2024-10-22/

**Comment by rpahl** [comment]
I recommend to not communicate the for-loop as the go-to approach for working with lists. Instead it is recommend to use one of the many functions working on lists. To loop through and to find numbers or indices, this can look as follows: 
```
numbers_list <- list(1, 2, 3, 4, 5)
squared_numbers <- lapply(numbers_list, FUN = function(x) x^2)
Position(squared_numbers,  f = function(x) x > 10)
Find(squared_numbers,  f = function(x) x > 10)
```
These base functions also are usually very fast.

**Reply by spsanderson** [reply]
Thank you for sharing!
